### PR TITLE
Numpy compatible logaddexp2

### PR DESCRIPTION
**Pull request by zoeygxy**
## Description ##
- Added [np.logaddexp2](https://docs.scipy.org/doc/numpy/reference/generated/numpy.logaddexp2.html) and its docs for future rendering. Scalar calculation is supported. 
- Fixed zero-size tensor error in BinaryBroadcastBackwardUseIn (GPU version).

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added [np.logaddexp2](https://docs.scipy.org/doc/numpy/reference/generated/numpy.logaddexp2.html) and its docs for future rendering. Scalar calculation is supported. 
- [x] Fixed zero-size tensor error in BinaryBroadcastBackwardUseIn.

## Comments ##
- The previous error case description: when one operand is non-zero-size (eg. shape of (1, 3)) and the other is zero-size (eg. shape of (2, 0, 3)), the backward calculation on GPU will raise a floating point error. Temporarily fixed in this PR as well as PR numpy compatible logaddexp.

### CODE REVIEW

**Comment by zoeygxy** [comment]
@haojin2

**Comment by zoeygxy** [comment]
Passed all CI tests on August 21st.

File: python/mxnet/ndarray/numpy/_op.py [hunk 1]
```diff
@@ -732,3 +733,66 @@ def get_list(arrays):
 
     arrays = get_list(arrays)
     return _npi.stack(*arrays, axis=axis, out=out)
+
+@set_module('mxnet.ndarray.numpy')
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: Move this up to after line 350, after `power`, remember the 2-blank-line gap between Python functions.

File: python/mxnet/numpy/multiarray.py [hunk 2]
```diff
@@ -1898,3 +1898,67 @@ def stack(arrays, axis=0, out=None):
     stacked : ndarray
         The stacked array has one more dimension than the input arrays."""
     return _mx_nd_np.stack(arrays, axis=axis, out=out)
+
+
+@set_module('mxnet.numpy')
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: Same here and in `_symbol.py`

File: src/operator/numpy/np_elemwise_broadcast_op.cc [hunk 3]
```diff
@@ -182,5 +202,14 @@ MXNET_OPERATOR_REGISTER_NP_BINARY_SCALAR(_npi_rpower_scalar)
...
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseOut{"_backward_rpower_scalar"});
 
+MXNET_OPERATOR_REGISTER_NP_BINARY_SCALAR(_npi_logaddexp2_scalar)
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::Compute<cpu, mshadow_op::logaddexp2>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_logaddexp2_scalar"});
+
+MXNET_OPERATOR_REGISTER_BINARY_SCALAR(_backward_logaddexp2_scalar)
+.set_attr_parser([](NodeAttrs *attrs) { attrs->parsed = std::stod(attrs->dict["scalar"]); })
+.set_attr<FCompute>("FCompute<cpu>", BinaryScalarOp::Backward<cpu,
+                                     mshadow_op::logaddexp2_grad_left>);
```
Comment (haojin2) [inline src/operator/numpy/np_elemwise_broadcast_op.cc]:
```c++
.set_attr<FCompute>("FCompute<cpu>",
                    BinaryScalarOp::Backward<cpu, mshadow_op::logaddexp2_grad_left>);
```

Comment (zoeygxy) [reply src/operator/numpy/np_elemwise_broadcast_op.cc]: Fixed. Thx!

File: python/mxnet/numpy/multiarray.py (hunk 2 above)
Comment (zoeygxy) [reply python/mxnet/numpy/multiarray.py]: Fixed. Thx!

File: python/mxnet/ndarray/numpy/_op.py (hunk 1 above)
Comment (zoeygxy) [reply python/mxnet/ndarray/numpy/_op.py]: Fixed. Thx!

**Pull request closed without merging**